test/lib/net/dhcp_test.rb

```diff
@@ -92,16 +92,28 @@ class DhcpTest < ActiveSupport::TestCase
     refute record1.valid?
   end
 
-  test "dhcp record validation should fail on MAC conflict" do
+  test "dhcp record must not validate when there is IP conflict" do
+    ProxyAPI::Resource.any_instance.stubs(:get).with("127.0.0.0/aa:bb:cc:dd:ee:01").returns(@lease1)
     ProxyAPI::Resource.any_instance.stubs(:get).with("127.0.0.0/127.0.0.1").returns(@lease1)
+    ProxyAPI::Resource.any_instance.stubs(:get).with("127.0.0.0/aa:bb:cc:dd:ee:02").raises(RestClient::ResourceNotFound, 'Record not found')
     record1 = Net::DHCP::Record.new(:hostname => "discovered_host1", :mac => "aa:bb:cc:dd:ee:02",
                                     :network => "127.0.0.0", :ip => "127.0.0.1",
                                     "proxy" => subnets(:one).dhcp_proxy)
     refute record1.conflicts.empty?
```

Comment:
the test checks that conflicts are not empty, but actually doesn't check for validity (which test description implies)

I'd add a ```refute record1.valid?``` just for conforming with description. same in the line 112